torch/fx/experimental/proxy_tensor.py

```diff
@@ -62,21 +63,20 @@ def is_sym_node(node):
     return "val" in node.meta and isinstance(node.meta['val'], py_sym_types)
 
 def set_proxy_slot(obj, tracer, proxy):
-    assert isinstance(obj, (torch.Tensor, SymNode)), type(obj)
-    d = obj.__dict__.setdefault(proxy_slot, weakref.WeakKeyDictionary())  # type: ignore[call-overload]
-    assert isinstance(d, weakref.WeakKeyDictionary)
-    # NB: Never clobber pre-existing proxy.  Although the proxies
-    # are in principle equivalent, when we do graph partitioning
-    # we need there not to be spurious dependencies on tangent inputs.
-    # This works because primals get their SymInts set first, and
-    # THEN later we allocate tangent inputs.  Make sure if a SymInt
-    # is derivable from a primal that we use that.
-    #
-    # However, we DO want to clobber proxies whenever we run an inplace operation
-    # on a tensor, and it affects the metadata on the proxy.
-    # This doesn't really apply to SymInts/SymFloats though, which are immutable.
-    if tracer not in d or isinstance(obj, torch.Tensor):
-        d[tracer] = proxy
+    if isinstance(obj, torch.Tensor):
+        # We DO want to clobber proxies whenever we run an inplace operation
```

Comment:
How do you know this was an inplace op?

Reply:
I think it doesn't matter (this works in all cases):

(1) if it's an out of place op, then we're setting a proxy on a fresh tensor (which doesn't have a proxy yet), so no clobbering happens.

(2) if it's an inplace op, then we're clobbering the proxy on an existing tensor.

I guess a useful sanity assert would be to check that there's already a proxy set when we see an inplace op, and not otherwise.